homeassistant/components/myq/cover.py

```diff
@@ -109,19 +141,41 @@ def unique_id(self):
 
     async def async_close_cover(self, **kwargs):
         """Issue close command to cover."""
+        self._last_action_timestamp = time.time()
         await self._device.close()
-        # Writes closing state
-        self.async_write_ha_state()
+        self._async_schedule_update_for_transition()
 
     async def async_open_cover(self, **kwargs):
         """Issue open command to cover."""
+        self._last_action_timestamp = time.time()
         await self._device.open()
-        # Writes opening state
+        self._async_schedule_update_for_transition()
+
+    @callback
+    def _async_schedule_update_for_transition(self):
         self.async_write_ha_state()
 
+        # Cancel any previous updates
+        if self._scheduled_transition_update:
+            self._scheduled_transition_update()
+
+        # Schedule an update for when we expect the transition
+        # to be completed so the garage door or gate does not
+        # seem like its closing or opening for a long time
+        self._scheduled_transition_update = async_track_point_in_time(
```

Comment:
We can use `async_call_later` instead, also in event helpers.

Reply:
Perfect. retested, works great. pushing shortly after some more testing

Reply:
Confirmed, tested enough that the neighbors are now wondering why the garage keeps going up and down.